ouroboros-consensus/test-consensus/Test/Dynamic/Network.hs

```diff
@@ -365,9 +414,21 @@ runNodeNetwork numCoreNodes numSlots slotLengths nodeJoinPlan nodeTopology
             nodeInfoDBs
       (chainDB, _) <- ChainDB.openDBInternal chainDbArgs True
 
+      -- We have a thread (see below) that forges EBBs for tests that involve
+      -- them. This variable holds the slot of the next EBB to be forged.
+      --
+      -- Even if the test doesn't involve EBBs, that thread must advance this
+      -- variable in order to unblock the node's block production thread.
+      nextEbbSlotVar <- uncheckedNewTVarM 0
+
       let nodeArgs = NodeArgs
             { tracers             = nullDebugTracers
-                { forgeTracer = nodeEventsForges nodeInfoEvents
+                { forgeTracer       = Tracer $ \case
+                    TraceForgeAboutToLead s -> do
+                      atomically $ do
+                        lim <- readTVar nextEbbSlotVar
+                        check $ s < lim
+                    o -> traceWith (nodeEventsForges nodeInfoEvents) o
```

Comment:
What if we want to make EBBs optional? Or even stop producing after some epoch?
It doesn't have to be part of this PR, but it would nice to be able to do that in the future.

Reply:
If `runNodeNetwork` is invoked with `NothingForgeEBB` then there will be no EBBs, but this `nextEbbSlotVar` will still increment -- there's a comment saying as much at the binding site of `nextEbbSlotVar`. It's slightly off-putting, but seemed preferable to having multiple `Maybe`s, etc.

No support for stopping EBBs at a certain slot yet.